### PR TITLE
BAH-1664-Updated to save labreports from labentry module in uploaded_results folder

### PR DESCRIPTION
**Pull request by deepthi-mantena**

### CODE REVIEW

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/PatientDocumentServiceImpl.java [hunk 1]
```diff
@@ -85,6 +87,16 @@ public String saveDocument(Integer patientId, String encounterTypeName, String c
...
+        if(encounterTypeName.equalsIgnoreCase(LAB_RESULT_ENCOUNTER_TYPE)){
+            basePath = BahmniCoreProperties.getProperty("bahmnicore.documents.lablitebaseDirectory");
+        } else{
+            basePath = getBasePath();
+        }
+        return basePath;
+    }
+
     private String getBasePath() {
         return BahmniCoreProperties.getProperty("bahmnicore.documents.baseDirectory");
```
Comment (mohan-13) [inline bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/PatientDocumentServiceImpl.java:101]: Can we move this line as well to the above function ? And we remove `getBasePath` function ?

**Review by mohan-13 — APPROVED** [review]

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/PatientDocumentServiceImpl.java (hunk 1 above)
Comment (deepthi-mantena) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/PatientDocumentServiceImpl.java:101]: @mohan-13 getBasepath is being used in two other functions so did not remove that.

**Review by abinaya-u — APPROVED** [review]

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/PatientDocumentServiceImpl.java [hunk 2]
```diff
@@ -45,6 +45,7 @@ public class PatientDocumentServiceImpl implements PatientDocumentService {
     private final Integer NO_OF_PATIENT_FILE_IN_A_DIRECTORY = 100;
     private final String VIDEO_FILE_TYPE = "video";
     private final String IMAGE_FILE_TYPE = "image";
+    public static final String LAB_RESULT_ENCOUNTER_TYPE = "LAB_RESULT";
```
Comment (angshu) [inline bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/PatientDocumentServiceImpl.java:48]: private ?

Comment (deepthi-mantena) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/PatientDocumentServiceImpl.java:48]: Sure Angshu will change

**Review by angshu — APPROVED** [review]

**Pull request merged**